shell/platform/windows/platform_view_manager.h

```diff
@@ -0,0 +1,76 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_WINDOWS_PLATFORM_VIEW_MANAGER_H_
+#define FLUTTER_SHELL_PLATFORM_WINDOWS_PLATFORM_VIEW_MANAGER_H_
+
+#include <functional>
+#include <map>
+#include <memory>
+#include <optional>
+
+#include "flutter/shell/platform/common/client_wrapper/include/flutter/method_channel.h"
+#include "flutter/shell/platform/windows/public/flutter_windows.h"
+#include "flutter/shell/platform/windows/task_runner.h"
+
+namespace flutter {
+
+// Possible reasons for change of keyboard foucs.
+enum class FocusChangeDirection {
+  kProgrammatic,  // Un-directed focus change.
+  kForward,       // Keyboard focus moves forwards, e.g. TAB-key.
+  kBackward       // Keyboard focus moves backwards, e.g. Shift+TAB.
+};
+
+// Keeps track of registered platform view types and platform view instances,
+// and is responsible for processing and responding to platform view related
+// method invokations from the framework.
+class PlatformViewManager {
+ public:
+  PlatformViewManager(TaskRunner* task_runner,
+                      BinaryMessenger* binary_messenger);
+
+  virtual ~PlatformViewManager();
+
+  // Add a new platform view instance to be lazily instantiated when it is next
+  // composited.
+  virtual void QueuePlatformViewCreation(PlatformViewId id,
+                                         std::string_view type_name);
+
+  // Create a queued platform view instance.
+  void InstantiatePlatformView(PlatformViewId id);
```

Comment:
Consider adding a comment that `id` must be already registered using `QueuePlatformViewCreation`.